app/Http/Controllers/CommentController.php

```diff
@@ -15,117 +18,140 @@ public function index(Post $post)
-        Storage::disk('public')->delete($comment->image_path);
-        $commentData['image_path'] = null;
-    }
+        $commentData = [
+            'temat' => $request->input('temat'),
+            'user_id' => $user_id,
+        ];
 
-    if ($request->hasFile('image')) {
-        if ($comment->imageExists()) {
+        if ($request->has('remove_image') && $comment->imageExists()) {
             Storage::disk('public')->delete($comment->image_path);
+            $commentData['image_path'] = null;
         }
 
-        $imagePath = $request->file('image')->store('comments', 'public');
-        $commentData['image_path'] = $imagePath;
-    }
+        if ($request->hasFile('image')) {
+            if ($comment->imageExists()) {
+                Storage::disk('public')->delete($comment->image_path);
+            }
+
+            $imagePath = $request->file('image')->store('comments', 'public');
+            $commentData['image_path'] = $imagePath;
+        }
 
-    $post = $comment->post;
+        $post = $comment->post;
 
-    $comment->update($commentData);
+        $comment->update($commentData);
 
-    return redirect()->route('comments.index', $post->id)->with('success', 'Komentarz został zaktualizowany.');
-}
+        return redirect()->route('comments.index', $post->id)->with('success', ' Comment update.');
+    }
 
+    public function destroy(Post $post, Comment $comment): JsonResponse
+    {
 
-public function destroy(Comment $comment): JsonResponse
-{
-    try {
-        if ($comment->imageExists()) {
-            Storage::disk('public')->delete($comment->image_path);
-        }
+        $user = Auth::user();
 
-        $comment->likes()->delete();
+        if ($user && ($user->isAdmin() || $user->id === $comment->user_id)) {
+            try {
+                if ($comment->imageExists()) {
+                    Storage::disk('public')->delete($comment->image_path);
+                }
 
-        $comment->delete();
+                $comment->likes()->delete();
 
-        Session::flash('success', 'Udało się usunąć comment.');
+                $comment->delete();
 
-        return response()->json([
-            'status' => 'success'
-        ]);
-    } catch (\Exception $e) {
-        \Log::error($e);
+                Session::flash('success', 'Comment destroy.');
 
-        if ($comment->imageExists()) {
-            \Log::info('Error occurred. Not deleting image from storage.');
+                return response()->json([
+                    'status' => 'success',
+                ]);
+            } catch (\Exception $e) {
+                \Log::error($e);
+
+                if ($comment->imageExists()) {
+                    \Log::info('Error occurred. Not deleting image from storage.');
+                }
+
+                return response()->json([
+                    'status' => 'error',
+                    'message' => 'Error!',
+                ])->setStatusCode(500);
+            }
```

Comment:
wywal tu tego ifa, jak jest error to go zapisz całego do loga, a tak to trzeba z fusów wróżyć, bo dev se w logach zapisał "Error occurred"